### ISSUE TITLE
Missing systemd default support / warning message for legacy sessions

### ISSUE THREAD

**Issue by vikaig** [Smithay/smithay#139]
After launch `./anvil --tty-udev` the background appears and the mouse pointer in the upper left corner, but everything hangs, the cursor does not move, Caps Lock does not work, CTRL-ALT-F1-12 does not work, you can exit this state only by pressing the reset button or ALT-SysRq-R
distro: Arch Linux
kernel: 5.0.13-arch1-1-ARCH
video: intel hd graphics
log:
> May 14 18:14:14.074 INFO Starting anvil on a tty using udev
May 14 18:14:14.076 INFO Listening on wayland socket, name: wayland-0
May 14 18:14:14.076 INFO Trying to create tty session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.076 INFO Running from tty: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.076 DEBG Opening device: "/dev/dri/card0", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.076 INFO DrmDevice initializing, smithay_module: backend_drm
May 14 18:14:14.076 WARN Unable to become drm master, assuming unpriviledged mode, smithay_module: backend_drm
May 14 18:14:14.098 DEBG Creating gbm device, smithay_module: backend_gbm
May 14 18:14:14.105 DEBG Creating egl context from device, smithay_module: backend_egl
May 14 18:14:14.105 DEBG Trying to initialize EGL with OpenGLES 3.0, smithay_module: renderer_egl, smithay_module: backend_egl
May 14 18:14:14.107 DEBG EGL No-Display Extensions: ["EGL_EXT_device_base", "EGL_EXT_device_enumeration", "EGL_EXT_device_query", "EGL_EXT_platform_base", "EGL_KHR_client_get_all_proc_addresses", "EGL_EXT_client_extensions", "EGL_KHR_debug", "EGL_EXT_platform_wayland", "EGL_EXT_platform_x11", "EGL_MESA_platform_gbm", "EGL_MESA_platform_surfaceless"], smithay_module: renderer_egl, smithay_module: backend_egl
May 14 18:14:14.107 INFO EGL Initialized, smithay_module: renderer_egl, smithay_module: backend_egl
May 14 18:14:14.107 INFO EGL Version: (1, 4), smithay_module: renderer_egl, smithay_module: backend_egl
May 14 18:14:14.107 INFO EGL Extensions: ["EGL_ANDROID_blob_cache", "EGL_ANDROID_native_fence_sync", "EGL_EXT_buffer_age", "EGL_EXT_create_context_robustness", "EGL_EXT_image_dma_buf_import", "EGL_EXT_image_dma_buf_import_modifiers", "EGL_IMG_context_priority", "EGL_KHR_config_attribs", "EGL_KHR_create_context", "EGL_KHR_create_context_no_error", "EGL_KHR_fence_sync", "EGL_KHR_get_all_proc_addresses", "EGL_KHR_gl_colorspace", "EGL_KHR_gl_renderbuffer_image", "EGL_KHR_gl_texture_2D_image", "EGL_KHR_gl_texture_3D_image", "EGL_KHR_gl_texture_cubemap_image", "EGL_KHR_image", "EGL_KHR_image_base", "EGL_KHR_image_pixmap", "EGL_KHR_no_config_context", "EGL_KHR_reusable_sync", "EGL_KHR_surfaceless_context", "EGL_EXT_pixel_format_float", "EGL_KHR_wait_sync", "EGL_MESA_configless_context", "EGL_MESA_drm_image", "EGL_MESA_image_dma_buf_export", "EGL_WL_bind_wayland_display", ""], smithay_module: renderer_egl, smithay_module: backend_egl
May 14 18:14:14.107 INFO Selected color format: PixelFormat { hardware_accelerated: true, color_bits: 24, alpha_bits: 8, depth_bits: 24, stencil_bits: 8, stereoscopy: false, double_buffer: true, multisampling: None, srgb: false }, smithay_module: renderer_egl, smithay_module: backend_egl
May 14 18:14:14.108 DEBG EGL context successfully created, smithay_module: renderer_egl, smithay_module: backend_egl
May 14 18:14:14.109 INFO EGL context created, smithay_module: renderer_egl, smithay_module: backend_egl
May 14 18:14:14.131 INFO Connected: DisplayPort
May 14 18:14:14.131 INFO Initializing EglSurface, smithay_module: backend_egl
May 14 18:14:14.132 INFO Initializing GbmSurface, smithay_module: backend_gbm
May 14 18:14:14.152 DEBG EGL surface successfully created, smithay_module: renderer_egl, smithay_module: backend_egl
May 14 18:14:14.161 DEBG Importing cursor, crtc: Handle(47), smithay_module: backend_gbm
May 14 18:14:14.162 DEBG Commiting new state, crtc: Handle(47), smithay_module: backend_gbm
May 14 18:14:14.162 DEBG Setting screen, crtc: Handle(47), smithay_module: backend_drm
May 14 18:14:14.170 INFO Initializing a xkbcommon handler with keymap query, options: None, variant: , layout: , model: , rules: , smithay_module: xkbcommon_handler, seat_name: seat0, smithay_module: seat_handler
May 14 18:14:14.172 INFO Loaded Keymap, name: English (US), smithay_module: xkbcommon_handler, seat_name: seat0, smithay_module: seat_handler
May 14 18:14:14.173 INFO Creating new wl_output, name: Drm, smithay_module: output_handler
May 14 18:14:14.176 DEBG Opening device: "/dev/input/event2", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.176 DEBG Opening device: "/dev/input/event9", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.176 DEBG Opening device: "/dev/input/event1", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.177 DEBG Opening device: "/dev/input/event0", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.177 DEBG Opening device: "/dev/input/event3", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.177 DEBG Opening device: "/dev/input/event4", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.177 DEBG Opening device: "/dev/input/event5", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.178 DEBG Opening device: "/dev/input/event6", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.178 DEBG Opening device: "/dev/input/event10", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.178 DEBG Opening device: "/dev/input/event11", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.178 DEBG Opening device: "/dev/input/event12", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.178 DEBG Opening device: "/dev/input/event8", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.179 DEBG Opening device: "/dev/input/event7", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 18:14:14.179 INFO Initializing a libinput backend, smithay_module: backend_libinput
May 14 18:14:14.179 INFO New input handler set, smithay_module: backend_libinput

**Comment by Drakulix** [comment]
This is likely not the complete log, but everything anvil is able to log before crashing.
Can you try to replace [this line](https://github.com/Smithay/smithay/blob/master/anvil/src/main.rs#L35) with: `::std::sync::Mutex::new(slog_term::term_full().fuse()).fuse()`?
Also please run the whole thing using: `env RUST_BACKTRACE=full ./anvil --tty-udev > error.log`

**Comment by vikaig** [comment]
This is a full log, the process does not crash, it continues to work after a hang, after ALT-SysRq-R CTRL-C the process normally ends, but with the code return - 130
After editing the line and recompiling the log has not changed:
`env RUST_BACKTRACE=full ./anvil --tty-udev > error.log 2>&1`
> May 14 20:31:41.811 INFO Starting anvil on a tty using udev
May 14 20:31:41.811 INFO Listening on wayland socket, name: wayland-0
May 14 20:31:41.811 INFO Trying to create tty session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.811 INFO Running from tty: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.812 DEBG Opening device: "/dev/dri/card0", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.813 INFO DrmDevice initializing, smithay_module: backend_drm
May 14 20:31:41.813 WARN Unable to become drm master, assuming unpriviledged mode, smithay_module: backend_drm
May 14 20:31:41.833 DEBG Creating gbm device, smithay_module: backend_gbm
May 14 20:31:41.862 DEBG Creating egl context from device, smithay_module: backend_egl
May 14 20:31:41.863 DEBG Trying to initialize EGL with OpenGLES 3.0, smithay_module: renderer_egl, smithay_module: backend_egl
May 14 20:31:41.867 DEBG EGL No-Display Extensions: ["EGL_EXT_device_base", "EGL_EXT_device_enumeration", "EGL_EXT_device_query", "EGL_EXT_platform_base", "EGL_KHR_client_get_all_proc_addresses", "EGL_EXT_client_extensions", "EGL_KHR_debug", "EGL_EXT_platform_wayland", "EGL_EXT_platform_x11", "EGL_MESA_platform_gbm", "EGL_MESA_platform_surfaceless"], smithay_module: renderer_egl, smithay_module: backend_egl
May 14 20:31:41.868 INFO EGL Initialized, smithay_module: renderer_egl, smithay_module: backend_egl
May 14 20:31:41.868 INFO EGL Version: (1, 4), smithay_module: renderer_egl, smithay_module: backend_egl
May 14 20:31:41.869 INFO EGL Extensions: ["EGL_ANDROID_blob_cache", "EGL_ANDROID_native_fence_sync", "EGL_EXT_buffer_age", "EGL_EXT_create_context_robustness", "EGL_EXT_image_dma_buf_import", "EGL_EXT_image_dma_buf_import_modifiers", "EGL_IMG_context_priority", "EGL_KHR_config_attribs", "EGL_KHR_create_context", "EGL_KHR_create_context_no_error", "EGL_KHR_fence_sync", "EGL_KHR_get_all_proc_addresses", "EGL_KHR_gl_colorspace", "EGL_KHR_gl_renderbuffer_image", "EGL_KHR_gl_texture_2D_image", "EGL_KHR_gl_texture_3D_image", "EGL_KHR_gl_texture_cubemap_image", "EGL_KHR_image", "EGL_KHR_image_base", "EGL_KHR_image_pixmap", "EGL_KHR_no_config_context", "EGL_KHR_reusable_sync", "EGL_KHR_surfaceless_context", "EGL_EXT_pixel_format_float", "EGL_KHR_wait_sync", "EGL_MESA_configless_context", "EGL_MESA_drm_image", "EGL_MESA_image_dma_buf_export", "EGL_WL_bind_wayland_display", ""], smithay_module: renderer_egl, smithay_module: backend_egl
May 14 20:31:41.870 INFO Selected color format: PixelFormat { hardware_accelerated: true, color_bits: 24, alpha_bits: 8, depth_bits: 24, stencil_bits: 8, stereoscopy: false, double_buffer: true, multisampling: None, srgb: false }, smithay_module: renderer_egl, smithay_module: backend_egl
May 14 20:31:41.872 DEBG EGL context successfully created, smithay_module: renderer_egl, smithay_module: backend_egl
May 14 20:31:41.872 INFO EGL context created, smithay_module: renderer_egl, smithay_module: backend_egl
May 14 20:31:41.897 INFO Connected: DisplayPort
May 14 20:31:41.897 INFO Initializing EglSurface, smithay_module: backend_egl
May 14 20:31:41.897 INFO Initializing GbmSurface, smithay_module: backend_gbm
May 14 20:31:41.917 DEBG EGL surface successfully created, smithay_module: renderer_egl, smithay_module: backend_egl
May 14 20:31:41.926 DEBG Importing cursor, crtc: Handle(47), smithay_module: backend_gbm
May 14 20:31:41.927 DEBG Commiting new state, crtc: Handle(47), smithay_module: backend_gbm
May 14 20:31:41.927 DEBG Setting screen, crtc: Handle(47), smithay_module: backend_drm
May 14 20:31:41.935 INFO Initializing a xkbcommon handler with keymap query, options: None, variant: , layout: , model: , rules: , smithay_module: xkbcommon_handler, seat_name: seat0, smithay_module: seat_handler
May 14 20:31:41.938 INFO Loaded Keymap, name: English (US), smithay_module: xkbcommon_handler, seat_name: seat0, smithay_module: seat_handler
May 14 20:31:41.939 INFO Creating new wl_output, name: Drm, smithay_module: output_handler
May 14 20:31:41.942 DEBG Opening device: "/dev/input/event2", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.943 DEBG Opening device: "/dev/input/event9", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.943 DEBG Opening device: "/dev/input/event1", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.943 DEBG Opening device: "/dev/input/event0", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.944 DEBG Opening device: "/dev/input/event3", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.944 DEBG Opening device: "/dev/input/event4", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.944 DEBG Opening device: "/dev/input/event5", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.945 DEBG Opening device: "/dev/input/event6", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.945 DEBG Opening device: "/dev/input/event10", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.945 DEBG Opening device: "/dev/input/event11", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.946 DEBG Opening device: "/dev/input/event12", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.946 DEBG Opening device: "/dev/input/event8", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.946 DEBG Opening device: "/dev/input/event7", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:41.947 INFO Initializing a libinput backend, smithay_module: backend_libinput
May 14 20:31:41.947 INFO New input handler set, smithay_module: backend_libinput
May 14 20:31:48.104 INFO Session shall become inactive., component: session_notifier, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.117 DEBG Session is now inactive, component: session_notifier, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.694 DEBG Session will become active again, component: session_notifier, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.700 DEBG Opening device: "/dev/input/event2", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.701 DEBG Opening device: "/dev/input/event9", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.701 DEBG Opening device: "/dev/input/event1", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.701 DEBG Opening device: "/dev/input/event0", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.702 DEBG Opening device: "/dev/input/event3", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.702 DEBG Opening device: "/dev/input/event4", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.703 DEBG Opening device: "/dev/input/event5", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.703 DEBG Opening device: "/dev/input/event6", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.704 DEBG Opening device: "/dev/input/event10", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.704 DEBG Opening device: "/dev/input/event11", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.704 DEBG Opening device: "/dev/input/event12", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.705 DEBG Opening device: "/dev/input/event8", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.705 DEBG Opening device: "/dev/input/event7", component: session, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 14 20:31:48.706 INFO Session is now active again, component: session_notifier, vt: 2, session_type: direct/vt, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto

**Comment by Drakulix** [comment]
> This is a full log, the process does not crash, it continues to work after a hang, after ALT-SysRq-R CTRL-C the process normally ends, but with the code return - 130
> After editing the line and recompiling the log did not change much:
> `env RUST_BACKTRACE=full ./anvil --tty-udev > error.log 2>&1`

Yeah, that is expected if the process is not crashing. That sounds a lot like a permissions problem. Are you not running systemd/logind? In that case, could you try running anvil using: `sudo -E ./anvil --tty-udev`? (`-E` preserves the environment because you need to have `XDG_RUNTIME_DIR` set, which is normally cleared by sudo).

**Comment by vikaig** [comment]
systemd and logind is running:
```
pstree -p | grep systemd
systemd(1)-+-NetworkManager(494)-+-{NetworkManager}(510)
           |-systemd(661)-+-(sd-pam)(662)
           |-systemd-journal(282)
           |-systemd-logind(495)
           |-systemd-udevd(302)
```
Running with `sudo -E` works fine, no input problems, there really was a permission problem, it is not clear just why it happens without sudo, systemd-logind normal works, as well as on purely installed Arch Linux

**Comment by Drakulix** [comment]
> systemd and logind is running:
> 
> ```
> pstree -p | grep systemd
> systemd(1)-+-NetworkManager(494)-+-{NetworkManager}(510)
>            |-systemd(661)-+-(sd-pam)(662)
>            |-systemd-journal(282)
>            |-systemd-logind(495)
>            |-systemd-udevd(302)
> ```
> 
> Running with `sudo -E` works fine, no input problems, there really was a permission problem, it is not clear just why it happens without sudo, systemd-logind normal works, as well as on purely installed Arch Linux

Looking at the log, anvil is selecting a legacy-type session `direct/vt` instead of a logind session, so that explains the permissions problem.

Anvil [does create](https://github.com/Smithay/smithay/blob/master/anvil/src/udev.rs#L100) an `AutoSession`, which [*should* log](https://github.com/Smithay/smithay/blob/master/src/backend/session/auto.rs#L101) any errors creating a logind-session, expect, if the `backend_session_logind` feature [is not enabled](https://github.com/Smithay/smithay/blob/master/src/backend/session/auto.rs#L119).

So that is likely the case. Try building anvil using: `cargo build --features backend_session_logind -p anvil`.
@vberger What was our reasoning not to include systemd in the default feature set?

**Comment by vberger** [comment]
Hmm, most likely I disabled it at some point working on smithay because it does not build on my system (I don't have systemd), and forgot about changing it properly....

**Comment by Drakulix** [comment]
> Hmm, most likely I disabled it at some point working on smithay because it does not build on my system (I don't have systemd), and forgot about changing it properly....

So the real issue then is, that `rust-systemd` should optionally use `libloading` or something similar so we can make systemd a true runtime dependency?

**Comment by vberger** [comment]
Well, I don't think that's absolutely necessary. I mean, it's not socking for a wayland compositor to have build time flags controlling its compatibility.

Personally I would be in favour of having a default set of build flags that would be rather conservative (meaning it should compile out of the box on most systems), and clearly document on the readme that a few cargo flags may be needed to get the full functionality.

Maybe we could add a few logging info message in the initialization of the direct session clarifying this as well. Something like "Falling back to the direct session, you need to enable the logind or elogind cargo feature at compilation for their support.", so that it all would be clear.

What do you think about that?

**Comment by vikaig** [comment]
Thanks for the help, after recompiling with `--features backend_session_logind` everything works fine, hopefully in the future systemd support will be enabled by default or the message:
> Falling back to the direct session, you need to enable the login or elogind cargo feature at compilation for their support.

P. S. When you exit there is a small error with the permissions:
> May 15 22:38:00.278 INFO Quitting.
May 15 22:38:00.279 INFO Closing logind session, vt: None, seat: seat0, id: 4, session_type: logind, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto
May 15 22:38:00.330 DEBG Surfaces dropped
May 15 22:38:00.330 DEBG Dropping device
May 15 22:38:00.331 INFO Dropping device: Some("/dev/dri/card0"), smithay_module: backend_drm
May 15 22:38:00.331 ERRO Failed to reset crtc (Handle(47)). Error: EACCES: Permission denied, smithay_module: backend_drm
May 15 22:38:00.331 INFO Closing logind session, vt: None, seat: seat0, id: 4, session_type: logind, smithay_module: backend_session, session_type: auto, smithay_module: backend_session_auto

**Comment by Drakulix** [comment]
Thanks for confirming. We will take care of this either way.

**Comment by rokit** [comment]
I'm having the same issue only I'm new to linux and not sure I understand everything in this thread. I ran this command in the `smithay` folder:
`cargo build --features backend_session_logind -p anvil` (built fine)
switched to another tty `ctrl-alt-f2`
cd'd into `smithay` again
ran `sudo -E ./anvil --tty-udev` (`sudo: ./anvil: command not found`)

Since that didn't work, I ran the command from the readme:
`cargo run -p anvil -- --tty-udev`

This launches anvil but hangs the system. The only thing I didn't do was change that line of code:
`::std::sync::Mutex::new(slog_term::term_full().fuse()).fuse()`. Is that part necessary still or am I missing something else?
